src/library_manager/lib_manager.c

```diff
@@ -509,98 +519,98 @@ static int lib_manager_load_data_from_host(struct lib_manager_dma_ext *dma_ext,
 	if (ret < 0)
 		goto return_on_fail;
 
-	dcache_invalidate_region((void *)dma_ext->dmabp.addr, size);
+	dcache_invalidate_region((void __sparse_cache *)dma_ext->dmabp.addr, size);
 
 return_on_fail:
 	rfree(dma_block_cfg);
 	return ret;
 }
 
-static int lib_manager_store_data(struct lib_manager_dma_ext *dma_ext, void *dst_addr,
-				  uint32_t dst_size)
+static int lib_manager_store_data(struct lib_manager_dma_ext *dma_ext,
+				  void __sparse_cache *dst_addr, uint32_t dst_size)
 {
+	struct lib_manager_dma_buf *const dma_buf = &dma_ext->dmabp;
 	uint32_t copied_bytes = 0;
-	uint32_t bytes_to_copy = 0;
-	struct lib_manager_dma_buf *dma_buf = &dma_ext->dmabp;
-	int ret;
 
 	while (copied_bytes < dst_size) {
+		uint32_t bytes_to_copy;
+		int ret;
+
 		if ((dst_size - copied_bytes) > MAN_MAX_SIZE_V1_8)
 			bytes_to_copy = MAN_MAX_SIZE_V1_8;
 		else
-			bytes_to_copy = (dst_size - copied_bytes);
+			bytes_to_copy = dst_size - copied_bytes;
```

Comment:
Unrelated changes creeping in.

Reply:
ok, I could split this into 2: a clean up and sparse stuff, would that help?

Reply:
@lyakh you can keep it, it's just a bit distracting to have these in bigger patches.

Reply:
It's not just a reviewer distraction, it also means no validation will never test these separately.